src/VirtoCommerce.PricingModule.Web/Module.cs

```diff
@@ -142,9 +141,12 @@ public void PostInitialize(IApplicationBuilder appBuilder)
             }
 
             var inProcessBus = appBuilder.ApplicationServices.GetService<IHandlerRegistrar>();
-            inProcessBus.RegisterHandler<PriceChangedEvent>(async (message, token) => await appBuilder.ApplicationServices.GetService<LogChangesChangedEventHandler>().Handle(message));
-            inProcessBus.RegisterHandler<ProductChangedEvent>(async (message, token) => await appBuilder.ApplicationServices.GetService<DeletePricesProductChangedEventHandler>().Handle(message));
-            inProcessBus.RegisterHandler<PriceChangedEvent>(async (message, token) => await appBuilder.ApplicationServices.GetService<IndexPricesProductChangedEventHandler>().Handle(message));
+            if (settingsManager.GetValue("Pricing.Search.EventBasedIndexation.Enable", false))
```

Comment:
here a rude mistake, why the  `Price.Search.EventBasedIndexation.Enable` setting manages of the other event handlers not related to the indexation process?